server/src/main/java/org/jboss/as/server/moduleservice/ServiceModuleLoader.java

```diff
@@ -191,22 +201,22 @@ public static void addService(final ServiceTarget serviceTarget, final Bootstrap
     /**
      * Returns the corresponding ModuleSpec service name for the given module.
      *
-     * @param identifier The module identifier
+     * @param name The module name
      * @return The service name of the ModuleSpec service
      */
-    public static ServiceName moduleSpecServiceName(ModuleIdentifier identifier) {
-        if (!isDynamicModule(identifier)) {
-            throw ServerLogger.ROOT_LOGGER.missingModulePrefix(identifier, MODULE_PREFIX);
+    public static ServiceName moduleSpecServiceName(String name) {
+        if (!isDynamicModule(name)) {
+            throw ServerLogger.ROOT_LOGGER.missingModulePrefix(name, MODULE_PREFIX);
         }
-        return MODULE_SPEC_SERVICE_PREFIX.append(identifier.getName()).append(identifier.getSlot());
+        return MODULE_SPEC_SERVICE_PREFIX.append(name);
     }
```

Comment:
I would say we should Keep the `ServiceName moduleSpecServiceName(ModuleIdentifier identifier)` variant and deprecate it for removal.

Reply:
nitpick: I would also keep the argument name as `identifier`. Even being a string, it will be an identifier (name+slot). It is not a hard requirement, but I think it is more coherent with what it is. The ModuleIdentifier class keeps the two fields (name and slot). It also has a method named moduleIdentifier.getName() which will return only the name, without a slot. So, `name` the argument as `name` here could be misleading and make someone think that it should be invoked with `moduleIdentifier.getName()`.